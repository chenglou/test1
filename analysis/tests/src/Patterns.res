module A = {
  let makeX = () => (1, 2)

  let (xxx, yyy) = makeX()

  type t = {name:string, age:int}
  
  let makeT = () => {name:"", age:0}

  let {name, age} = makeT()

  let (a | a, b) = makeX()

  let [v1,v2,v3] = [1,2,3]
}

let y = A.xxx
//        ^def

let z = A.yyy

let n = A.name
//         ^def

let n = A.a
//        ^def

let n = A.v1
//        ^def
