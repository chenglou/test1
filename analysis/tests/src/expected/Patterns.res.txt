Definition tests/src/Patterns.res 16:10
{"uri": "Patterns.res", "range": {"start": {"line": 3, "character": 7}, "end": {"line": 3, "character": 10}}}

Definition tests/src/Patterns.res 21:11
{"uri": "Patterns.res", "range": {"start": {"line": 9, "character": 7}, "end": {"line": 9, "character": 11}}}

Definition tests/src/Patterns.res 24:10
{"uri": "Patterns.res", "range": {"start": {"line": 11, "character": 7}, "end": {"line": 11, "character": 8}}}

Definition tests/src/Patterns.res 27:10
{"uri": "Patterns.res", "range": {"start": {"line": 13, "character": 7}, "end": {"line": 13, "character": 9}}}

